astropy/cosmology/core.py

```diff
@@ -157,6 +157,11 @@ def clone(self, **kwargs):
         if len(kwargs) == 0:
             return self
 
+        # There are changed parameter values.
+        # The name needs to be changed accordingly, if it wasn't already.
+        kwargs.setdefault("name", (self.name + " (modified)"
```

Comment:
I'm curious why you use "setdefault" here since the return value if the function is not used.
```suggestion
        kwargs['name'] = None if self.name is None else self.name + " (modified)"
```

Reply:
Note that this suggestion can't be applied directly, since the line below will also have to be removed.

Reply:
I used ``setdefault`` because it only sets the value if one is not already present. The suggested code doesn't quite work because it doesn't check if kwargs already has "name", in which case it should not be changed. An alternative one-liner is
``kwargs["name"] = kwargs.get("name", self.name + " (modified)" if self.name is not None else None)``.

Reply:
It was mostly just a lazy way of skipping an if-block to make getting that 100% codecov easier. This doesn't need multiple test cases. This is the equivalent code.

```
if "name" not in kwargs:
    kwargs['name'] = None if self.name is None else self.name + " (modified)"
```

Reply:
Great, thank you for the explanation. I did not know about "setdefault" before, I will definitely use that more often in the future myself!